pkg/controllers/csi/driver/server.go

```diff
@@ -233,25 +238,37 @@ func (svr *Server) NodeExpandVolume(context.Context, *csi.NodeExpandVolumeReques
 
 func logGRPC() grpc.UnaryServerInterceptor {
 	return func(ctx context.Context, req any, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (any, error) {
-		if info.FullMethod == "/csi.v1.Identity/Probe" || info.FullMethod == "/csi.v1.Node/NodeGetCapabilities" {
-			return handler(ctx, req)
-		}
+		var methodName string
 
-		methodName := ""
-
-		if info.FullMethod == "/csi.v1.Node/NodePublishVolume" {
+		switch {
+		case info.FullMethod == "/csi.v1.Node/NodePublishVolume":
 			req := req.(*csi.NodePublishVolumeRequest)
 			methodName = "NodePublishVolume"
 			log.Info("GRPC call", "method", methodName, "volume-id", req.GetVolumeId())
-		} else if info.FullMethod == "/csi.v1.Node/NodeUnpublishVolume" {
+		case info.FullMethod == "/csi.v1.Node/NodeUnpublishVolume":
 			req := req.(*csi.NodeUnpublishVolumeRequest)
 			methodName = "NodeUnpublishVolume"
 			log.Info("GRPC call", "method", methodName, "volume-id", req.GetVolumeId())
+		default:
+			resp, err := handler(ctx, req)
+			if err != nil {
+				log.Error(err, "GRPC failed", "full_method", info.FullMethod)
+			}
+
+			return resp, err
+		}
+
+		if counter.Load() > MaxGrpcRequests {
+			return nil, status.Error(codes.ResourceExhausted, fmt.Sprintf("rate limit exceeded, current value %d more than max %d", counter.Load(), MaxGrpcRequests))
 		}
 
+		counter.Add(1)
```

Comment:
Be aware that if you increase the counter after the if, the rate limiter will never be precise. 

It can happen that multiple go routines are already executing after the if and then increase the counter. Would put it in front to ensure that the request limit is actually enforced.

Reply:
@gkrenn  could you please be more percise? after which "if"?

Reply:
fixed in 87ab28812